x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/RollupAction.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.core.ilm;
+
+import org.elasticsearch.client.Client;
+import org.elasticsearch.cluster.metadata.IndexMetadata;
+import org.elasticsearch.common.ParseField;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.xcontent.ConstructingObjectParser;
+import org.elasticsearch.common.xcontent.ObjectParser;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.xpack.core.ilm.Step.StepKey;
+import org.elasticsearch.xpack.core.rollup.v2.RollupV2Config;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * A {@link LifecycleAction} which sets the index's priority. The higher the priority, the faster the recovery.
+ */
+public class RollupAction implements LifecycleAction {
+    public static final String NAME = "rollup";
+
+    @SuppressWarnings("unchecked")
+    private static final ConstructingObjectParser<RollupAction, Void> PARSER = new ConstructingObjectParser<>(NAME,
+        a -> new RollupAction((RollupV2Config) a[0]));
+
+    //package private for testing
+    final RollupV2Config config;
+
+    static {
+        PARSER.declareField(ConstructingObjectParser.constructorArg(),
+            (p, c) -> RollupV2Config.fromXContent(p, "_source_index_todo"),
```

Comment:
I think as it currently is, we'll have to rewrite the parsing, because we won't know what the source index name is until much later, and we won't know the destination index name until actual execution also.

I think we may want to not use `RollupV2Config`, but instead keep our own configuration option with the options we need. (At least if we want to keep `RollupV2Config` requiring the source and destination names up front

Reply:
We may also potentially want a way to have ILM delete the source index automatically after the rollup has been successfully completed, in case the user wants to do a destructive rollup

Reply:
great point on the note about the optional delete. will add